### ISSUE TITLE
Evaluate only on the SQA3D dataset

### ISSUE THREAD

**Issue by rickeyhhh** [embodied-generalist/embodied-generalist#57]
Thank you so much for sharing such great work!

I would like to know how it can be evaluated **on the SQA3D dataset only**? What file parameters need to be modified?
Looking forward to your guidance.

**Comment by huangjy-pku** [comment]
You can first modify the task config. For example, in [test_scannet.yaml](https://github.com/embodied-generalist/embodied-generalist/blob/main/configs/task/test_scannet.yaml) you can comment the lines regarding Scan2Cap and ScanQA.

After that, you can run the script with an additional argument `mode=eval` and specify `task=test_scannet`. You can check [readme](https://github.com/embodied-generalist/embodied-generalist?tab=readme-ov-file#running) for more information.

**Comment by rickeyhhh** [comment]
I modified the **inference.sh** file as follows:

> python launch.py --mode python \
                 --run_file inference.py \
                 --config configs/default.yaml \
                 mode=eval \
                 task=test_scannet \
                 note=sft_noact \
                 pretrained_ckpt_path=/mnt/data/user_files/.../embodied-generalist \

But after running it, it turned out to be a probe example:

> Loading checkpoint shards: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████| 2/2 [00:15<00:00,  7.84s/it]
[2025-04-01 16:51:41,557][model.leo_agent][INFO] - Build Vicuna7B from /mnt/data/user_files/.../embodied-generalist/vicuna-7b
[2025-04-01 16:51:41,560][model.leo_agent][INFO] - Freeze LLM
[2025-04-01 16:51:42,879][timm.models._builder][INFO] - Loading pretrained weights from file (/mnt/data/user_files/.../embodied-generalist/CLIP-convnext_base_w-laion2B-s13B-b82K/open_clip_pytorch_model.bin)
[2025-04-01 16:51:43,647][timm.models._helpers][INFO] - Loaded  from checkpoint '/mnt/data/user_files/.../embodied-generalist/CLIP-convnext_base_w-laion2B-s13B-b82K/open_clip_pytorch_model.bin'
[2025-04-01 16:51:43,768][model.vision2d][INFO] - Build Backbone2D: convnext_base_laion2b, pretrain = True, freeze = True, pooling = avg
[2025-04-01 16:51:43,817][model.pcd_backbone][INFO] - Build PointcloudBackbone: PointNetPP
[2025-04-01 16:51:43,830][model.pcd_backbone][INFO] - Load PointNetPP weights from /mnt/data/user_files/.../embodied-generalist/pointnetpp_vil3dref.pth
[2025-04-01 16:51:43,831][model.pcd_backbone][INFO] - Freeze PointNetPP
[2025-04-01 16:51:43,865][model.vision3d][INFO] - Build 3D module: OSE3D
[2025-04-01 16:51:43,899][model.leo_agent][INFO] - Apply LoRA with configs: {'flag': True, 'rank': 16, 'alpha': 16, 'target_modules': ['q_proj', 'k_proj', 'v_proj', 'o_proj', 'gate_proj', 'up_proj', 'down_proj'], 'dropout': 0.0}
[2025-04-01 16:53:10,958][__main__][INFO] - Probe: load model from /mnt/data/user_files/.../embodied-generalist
[2025-04-01 16:53:11,437][trainer.leo_trainer][INFO] - Successfully loaded from /mnt/data/user_files/.../embodied-generalist, load_model_only = True
  0%|                                                                                                                                        | 0/1 [00:00<?, ?it/s][2025-04-01 16:53:22,462][__main__][INFO] - {'source': '3rscan', 'scene_id': '0cac75d0-8d6f-2d13-8c26-d771a31c3f50', 'situation': '', 'instruction': 'Describe this scene.', 'response': 'In this room, there is a couch. It is upholstered and placed in front of the commode. There are also two armchair, one on each side of the couch. The armchair are padded and beige in color. On the left side of the couch, there is a coffee table. On the right side of the couch, there is an ottoman. Behind the couch, there is a wardrobe. On the left side of the couch, there is a shelf attached to the wall. On the right side of the couch, there is another shelf attached to the wall.'}
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:11<00:00, 11.01s/it]
[2025-04-01 16:53:23,658] - Launched

I want to evaluate on the entire SQA3D dataset and get the metric results. Could you please tell me how to do it?

**Comment by rickeyhhh** [comment]
I think it has something to do with the **scene_id** setting in **default.yaml**, which is forced to be overridden by scene_id every time the script is executed. So I still only get one output of **scene_ids: [0cac75d0-8d6f-2d13-8c26-d771a31c3f50]**.
Here is my default.yaml settings.

> defaults:
>   - data: default
>   - task: tuning_noact
>   - llm: vicuna7b
>   - vision2d: convnext
>   - vision3d: ose3d_pointnetpp
>   - _self_
> 
> # exp general info
> name: LEO-1   # project name of wandb
> note: default   # run name of wandb
> 
> rng_seed: 42
> num_gpu: 4   # will be overrided by launch.py
> mode: train
> naming_keywords: [note]   # choose keywords to feature the exp run dir
> base_dir: /mnt/data/user_files/mrq/embodied-generalist
> exp_dir: ""   # temporarily empty, will be set by run.py as base_dir + name + *naming_keywords
> pretrained_ckpt_path: ""   # specified on launch
> 
> logger:
>   name: wandb
>   entity: TBD
> 
> dataset_wrapper_args:
>   max_obj_len: 60
> 
> dataloader:
>   train:
>     batchsize: 4   # per-gpu batchsize
>     num_workers: 4
>   eval:
>     batchsize: 4   # per-gpu batchsize
>     num_workers: 4
> 
> trainer: LeoTrainer
> training:
>   epochs: ${task.training.epochs}
>   gradient_accumulation_steps: 5
>   grad_norm: 5.0
>   optim:
>     name: AdamW
>     args:
>       lr: ${task.training.lr}
>       betas: [0.9, 0.999]
>       weight_decay: 0.05
>   schedule:
>     name: linear_warmup_cosine_decay
>     args:
>       warmup_steps: 400
> 
> eval:
>   num_batch_val: 50
>   val_interval: 1
> 
> # model misc
> clip_txt_guidance:
>   flag: False   # for eai
>   clip_out_dim: 1024
> 
> # inference
> probe:
>   sources: [3rscan]
>   scene_ids: [0cac75d0-8d6f-2d13-8c26-d771a31c3f50]
>   situations: ""
>   instructions: [Describe this scene.]
>   save_obj_tokens: True
>

**Comment by huangjy-pku** [comment]
I mean, do not use `inference`. Instead, just use normal `launch` script and set `mode=eval`. **Do not** specify `--run_file inference.py`.
